src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java

```diff
@@ -161,14 +162,28 @@ private void clearIssueCacheAndPublishEmptyDiagnostics(URI f) {
 
   private void analyze(AnalysisTask task, Optional<WorkspaceFolderWrapper> workspaceFolder, Map<URI, VersionedOpenFile> filesToAnalyze) {
     if (workspaceFolder.isPresent()) {
-      // We can only have the same binding for a given folder
+
+      var notebooksToAnalyze = new HashMap<URI, VersionedOpenFile>();
+      var nonNotebooksToAnalyze = new HashMap<URI, VersionedOpenFile>();
+      filesToAnalyze.forEach((uri, file) -> {
+        if (openNotebooksCache.isNotebook(uri)) {
+          notebooksToAnalyze.put(uri, file);
+        } else {
+          nonNotebooksToAnalyze.put(uri, file);
+        }
+      });
```

Comment:
@sophio-japharidze-sonarsource I believe this is arguably more readable than the functional-style soup I pushed earlier 😅

Reply:
YES! 💯